protocol/x/prices/keeper/market_param.go

```diff
@@ -44,6 +44,13 @@ func (k Keeper) ModifyMarketParam(
 			errorsmod.Wrapf(types.ErrMarketExponentCannotBeUpdated, lib.UintToString(updatedMarketParam.Id))
 	}
 
+	// Validate base/quote uniqueness
```

Comment:
This doesn't seem necessary? Isn't there already a check that the updatedMarket must already exist in state + we've introduced a uniqueness constraint across all market-pairs in state?

Reply:
Looks like this check is actually necessary, as it's possible for a `MarketParam` update to change the marketParam.Pair, and the updated Pair may conflict with one in state